neuracore/ml/core/neuracore_model.py

```diff
@@ -115,6 +116,115 @@ def _validate_input_output_types(self) -> None:
                 f"{req_output_data_types - types_in_dataset}"
             )
 
+    def _populate_output_mapping(
+        self,
+        robot_to_output_mapping: dict[str, dict[DataType, list[str]]],
+        keys: dict[str, list[str]],
+        data_type: DataType,
+    ) -> dict[str, dict[DataType, list[str]]]:
+        """Populate the output mapping for a given data type.
+
+        Args:
+            robot_to_output_mapping: Output mapping per robot.
+            keys: Keys for each robot for the given data type.
+            data_type: Data type of the mapping.
+
+        Returns:
+            dict[str, dict[DataType, list[str]]]: Output mapping per robot.
+        """
+        for robot_name in keys.keys():
+            if robot_name not in robot_to_output_mapping:
+                robot_to_output_mapping[robot_name] = {}
+            robot_to_output_mapping[robot_name][data_type] = keys[robot_name]
+        return robot_to_output_mapping
+
+    def _get_robot_to_output_mapping(self) -> dict[str, dict[DataType, list[str]]]:
+        """Get the output mapping from the dataset description for each robot.
+
+        Return the output mapping of robots that have all requested output data types.
+
+        Returns:
+            dict[str, dict[DataType, list[str]]]: Output mapping per robot.
+        """
+        output_data_types = self.model_init_description.output_data_types
+        robot_to_output_mapping: dict[str, dict[DataType, list[str]]] = {}
+        if DataType.JOINT_TARGET_POSITIONS in output_data_types:
```

Comment:
Let's try and get rid of this giant if block; I know this already existed in a previous file, but worth getting rid of why we are here. 

at top of file, something like:
```
DATA_TYPE_TO_DATASET_DESCRIPTION_ATTR = {
  DataType.JOINT_POSITIONS: "joint_positions",
  .....
  }
 ```
 
 And then in this function:
 ```
 for dt in output_data_types:
   keys = getattr(self.dataset_description, DATA_TYPE_TO_DATASET_DESCRIPTION_ATTR[dt]).robot_to_ncdata_keys
   robot_to_output_mapping = self._populate_output_mapping( robot_to_output_mapping, keys,dt)
```

You may need a special case for 1 or two data types., e.g. CUSTOM